posthog/hogql/test/_test_parser.py

```diff
@@ -1348,13 +1348,13 @@ def test_select_limit_offset(self):
                 ),
             )
             self.assertEqual(
-                self._select("select 1 from events LIMIT 1 OFFSET 3 BY 1, event"),
+                self._select("select 1 from events LIMIT 1 BY event LIMIT 1 OFFSET 3"),
```

Comment:
This is the `LIMIT n BY expressions` variant. Let's also add test cases for the other two supported variants:
- `LIMIT offset_value, n BY expressions`
- `LIMIT n OFFSET offset_value BY expressions`